pkg/operator/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -526,94 +532,139 @@ func (r *ReconcileCredentialsRequest) Reconcile(ctx context.Context, request rec
+				if t.Reason() == minterv1.OrphanedCloudResource {
+					// not a critical error, just set the condition to communicate
+					// what happened
+					provisionErr = false
+				} else {
+					logger.Errorf("error syncing credentials: %v", syncErr)
+					provisionErr = true
+				}
 
-	if !cloudCredsSecretUpdated && !isStale && hasRecentlySynced && crSecretExists && !hasActiveFailureConditions && cr.Status.Provisioned {
-		logger.Debug("lastsyncgeneration is current and lastsynctimestamp was less than an hour ago, so no need to sync")
-		// Since we get no events for changes made directly to the cloud/platform, set the requeueAfter so that we at
-		// least periodically check that nothing out in the cloud/platform was modified that would require us to fix up
-		// users/permissions/tags/etc.
-		return reconcile.Result{RequeueAfter: defaultRequeueTime}, nil
-	}
+				cr.Status.Provisioned = !provisionErr
 
-	credsExists, err := r.Actuator.Exists(context.TODO(), cr)
-	if err != nil {
-		logger.Errorf("error checking whether credentials already exists: %v", err)
-		return reconcile.Result{}, err
-	}
+				logger.Errorf("errored with condition: %v", t.Reason())
+				r.updateActuatorConditions(cr, t.Reason(), syncErr)
+			default:
+				logger.Errorf("unexpected error while syncing credentialsrequest: %v", syncErr)
+				return reconcile.Result{}, syncErr
+			}
 
-	var syncErr error
-	if !credsExists {
-		syncErr = r.Actuator.Create(context.TODO(), cr)
+		} else {
+			// it worked so clear any actuator conditions if they exist
+			r.updateActuatorConditions(cr, "", nil)
+			cr.Status.Provisioned = true
+		}
 	} else {
-		syncErr = r.Actuator.Update(context.TODO(), cr)
-	}
-
-	var provisionErr bool
-	if syncErr != nil {
-		switch t := syncErr.(type) {
-		case actuator.ActuatorStatus:
-			if t.Reason() == minterv1.OrphanedCloudResource {
-				// not a critical error, just set the condition to communicate
-				// what happened
-				provisionErr = false
-			} else {
-				logger.Errorf("error syncing credentials: %v", syncErr)
-				provisionErr = true
-			}
+		credentialsRootSecret, err := r.Actuator.GetCredentialsRootSecret(context.TODO(), cr)
+		if err != nil {
+			log.WithError(err).Debug("error retrieving cloud credentials secret, admin can remove root credentials in mint mode")
+		}
+		cloudCredsSecretUpdated := credentialsRootSecret != nil && credentialsRootSecret.ResourceVersion != cr.Status.LastSyncCloudCredsSecretResourceVersion
+		isStale := cr.Generation != cr.Status.LastSyncGeneration
+		hasRecentlySynced := cr.Status.LastSyncTimestamp != nil && cr.Status.LastSyncTimestamp.Add(syncPeriod).After(time.Now())
+		hasActiveFailureConditions := checkForFailureConditions(cr)
+
+		if !cloudCredsSecretUpdated && !isStale && hasRecentlySynced && crSecretExists && !hasActiveFailureConditions && cr.Status.Provisioned {
+			logger.Debug("lastsyncgeneration is current and lastsynctimestamp was less than an hour ago, so no need to sync")
```

Comment:
nit: I think it might be nice to get all of the inputs to this `if` statement as structured fields in the logger - I can imagine a world where one of those conditions is not correct and the log message today does not provide enough information to figure out which

Reply:
Seems reasonable. Done in upcoming commit.